app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,626 @@
+}
+```
+
+We can obtain method handles for these transformation methods:
+
+```java
+MethodHandle negate = lookup.findStatic(Example.class, "negate", MethodType.methodType(boolean.class, boolean.class));
+MethodHandle increment = lookup.findStatic(Example.class, "increment", MethodType.methodType(int.class, int.class));
+```
+
+and use them to get a new method handle having filtered arguments:
+
+```java
+// applies filter 'increment' to argument at index 0, 'negate' to the last argument, 
+// and passes the result to 'targetMethodHandle'
+MethodHandle withFilters = MethodHandles.filterArguments(targetMethodHandle, 0, increment, null, null, negate);
+withFilters.invoke(3, "abc", 5L, false); // outputs "4abc5true"
+```
+
+### Fold arguments
+When we want to perform pre-processing of one or more arguments before the invocation of a `MethodHandle`, we
+can use [`MethodHandles.foldArguments`](javadoc:MethodHandles.foldArguments(MethodHandle,int,MethodHandle)) and provide
+it with the method handle of any combiner method which will accept arguments starting at any preferred position.
+
+Let's assume that we have a `target` method:
+
+```java
+private static void target(int ignored, int sum, int a, int b) {
+    System.out.printf("%s + %s equals %s and %s is ignored%n", a, b, sum, ignored);
+}
+```
+
+Using `foldArguments` we can pre-process subset of its arguments and insert the resulting value as another
+argument and proceed to execution of the `target` method.
+
+In our example, we have arguments `int a, int b` at the end. We can pre-process any amount of arguments, but they all
+must be at the end. Let's say, we would like to calculate a sum of these two values `a` and `b`, so let's create a method
+for that:
+
+```java
+private static int sum(int a, int b) {
+    return a + b;
+}
+```
+
+Where will the resulting value go exactly? It will be inserted into one of the arguments of our `target` method. It must
+be the argument right before the arguments that we are going to fold, so in our example argument `int sum`. The argument
+reserved for the folding result can't be at any other position. If the `target` method needs to accept more arguments not
+related to this folding logic, they all must go first.
+
+Let's create the method handles and see how we should combine them together:
+
+```java
+MethodHandle targetMethodHandle = lookup.findStatic(Example.class, "target",
+        MethodType.methodType(void.class, int.class, int.class, int.class, int.class));
+MethodHandle combinerMethodHandle = lookup.findStatic(Example.class, "sum",
+        MethodType.methodType(int.class, int.class, int.class));
+MethodHandle preProcessedArguments = MethodHandles.foldArguments(targetMethodHandle, 1, combinerMethodHandle);
+```
+
+The `foldArguments` method accepts:
+- Target method handle, in our case the one pointing to `target` method;
+- `int` number specifying the starting position of arguments related to folding. In our case, the `sum` argument
```

Comment:
As far as I see, you didn't mention the type of an argument like this anywhere else.